### PR TITLE
extensions: avoid external calls on DapsIntegrationTest

### LINKED ISSUE #256: Eliminate request to external DAPS from within integration test
The [DapsIntegrationTest](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/extensions/iam/daps/src/test/java/org/eclipse/dataspaceconnector/iam/daps/DapsIntegrationTest.java#L67) issues real requests to the external DAPS instance hosted by Fraunhofer AISEC. 
Although it's working, IMO it's bad practice and also obliges external conditions we cannot influence.

There is a pre-built docker image of the [DAPS omejdn-server](https://github.com/Fraunhofer-AISEC/omejdn-server/pkgs/container/omejdn-server). 
It'd be nice to have some volatile DAPS instance running during integration tests - possibly by using [testcontainers](https://www.testcontainers.org/quickstart/junit_5_quickstart/) from within the integration test itself.

<sub> Denis Neuling <denis.neuling@daimler.com>, Daimler TSS GmbH, [legal info/Impressum](https://github.com/Daimler/daimler-foss/blob/master/LEGAL_IMPRINT.md)

### PR DESCRIPTION
**Pull request by ndr-brt**
Fix #256

### CODE REVIEW

**Comment by ndr-brt** [comment]
@paullatzelsperger do you know why the `Daps-Integration-Test` job is not executed?

**Comment by paullatzelsperger** [comment]
All integration tests need to be added explicitly to the integrationtests.yaml file

**Comment by ndr-brt** [comment]
I added it: https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/pull/260/files#diff-65d9f8563af8ede2b36a1d6eae851bbef13b587ae19934044f5929df88c9613c
maybe I missed something, but can't figure out what

**Comment by paullatzelsperger** [comment]
> I added it: https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/pull/260/files#diff-65d9f8563af8ede2b36a1d6eae851bbef13b587ae19934044f5929df88c9613c maybe I missed something, but can't figure out what

ok, so the way I understand this, the reason is that the Github runner executes the tests against the `pull_request_target` which is `upstream/main`, where those i-tests do not yet exist (since they are only on your fork).

**Review by bscholtes1A — APPROVED** [review]

**Comment by jimmarino** [comment]
@ndr-brt, @paullatzelsperger  Is this ready to be merged?

**Comment by ndr-brt** [comment]
For me it's ok, the only thing is that this IT was never run on the github actions, so I think we need to merge to see if that works

File: extensions/iam/daps/src/test/resources/keys/OEI6REU6RUU6NEM6N0M6MkQ6N0I6OTM6NTE6MkY6RkQ6QTA6RDM6NTM6NEM6NTg6Q0E6OUQ6NzA6QzI6a2V5aWQ6Q0I6OEM6Qzc6QjY6ODU6Nzk6QTg6MjM6QTY6Q0I6MTU6QUI6MTc6NTA6MkY6RTY6NjU6NDM6NUQ6RTg=.cert [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+L2lkcy5jcmwwVAYIKwYBBQUHAQEESDBGMEQGCCsGAQUFBzAChjhodHRwOi8vZG93
+bmxvYWRzLmFpc2VjLmZyYXVuaG9mZXIuZGUvcm9vdGNhY2VydDIwMTYuY2VydDAd
+BgNVHSUEFjAUBggrBgEFBQcDAgYIKwYBBQUHAwEwFAYDVR0RBA0wC4IJbG9jYWxo
+b3N0MA0GCSqGSIb3DQEBCwUAA4IBAQA97zj5g64tEOPuY/3sCDykuDY12YR++ITG
+ZwC9lOhs2DyUX7dHMesgaEhm1WHFNsRuYzhzxyg3dBUM+T26ki0r3Mwckv/pBsw0
+5UEPEphPlTaovniZdcrUFpkic9ybfoobsUD1MKdJq72AQvmVHtXZRWbWi9tEpPw3
+rUhaNI8ItL+N40I75NnDpDtosJkrCheys7hjdsg7Qvx4jn4N1OZ6hd20cSwmlLJE
+keTfJ3DDyQ084enM4+ybMVlUK/jbg9n86Ej7X2H9hI48lgmEY7AzSB7HcK8OeIZO
+Eyq3wsFSqby7zcKcEs2ao46Uwmma1NMkjLAyZDNzfn+8VKJdfz+9
+-----END CERTIFICATE-----
```
Comment (denisneuling) [inline extensions/iam/daps/src/test/resources/keys/OEI6REU6RUU6NEM6N0M6MkQ6N0I6OTM6NTE6MkY6RkQ6QTA6RDM6NTM6NEM6NTg6Q0E6OUQ6NzA6QzI6a2V5aWQ6Q0I6OEM6Qzc6QjY6ODU6Nzk6QTg6MjM6QTY6Q0I6MTU6QUI6MTc6NTA6MkY6RTY6NjU6NDM6NUQ6RTg=.cert:25]: Just as a note: This cert is expiring within the next 2 years on Aug 16th 2023 - still a long time to invalidation, but we should make a note somewhere.

```
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number: 208 (0xd0)
        Signature Algorithm: sha256WithRSAEncryption
        Issuer: C = DE, O = Fraunhofer, CN = IDS Test SubCA 2018
        Validity
            Not Before: Aug 16 11:51:57 2021 GMT
            Not After : Aug 16 11:51:57 2023 GMT
        Subject: C = DE, O = IDS, OU = Plugfest 2021 08, CN = 6c173791-eda7-46a2-9ba7-f5b6eb36d6b1
        Subject Public Key Info:
            Public Key Algorithm: rsaEncryption
                RSA Public-Key: (2048 bit)
                Modulus:
                    00:c0:fd:5a:0e:28:77:0f:3d:f9:1e:28:0d:1e:82:
                    0b:d6:7e:14:2a:15:bc:a4:3f:cb:c3:40:9f:91:b6:
                    8e:17:9b:8d:91:9f:81:7d:32:d7:3e:c0:47:ea:39:
                    a8:5f:7b:d3:87:4f:32:64:ec:80:6f:3c:61:e6:16:
                    47:21:1f:75:7d:4c:c0:ac:55:bd:88:b1:a1:33:89:
                    fc:78:60:03:98:fc:67:85:43:ed:37:2f:a9:80:87:
                    b0:68:a7:0b:e8:5a:90:e9:b7:7b:a7:42:3c:ce:dc:
                    8a:b8:07:a0:2c:1f:49:4b:f6:32:c5:81:99:71:77:
                    ca:e0:58:9d:98:b9:89:f9:f1:64:a2:31:c8:c1:56:
                    6b:96:76:6e:ab:33:57:d4:08:a4:1f:5c:05:0a:fe:
                    01:51:3e:29:03:18:ea:4b:70:83:b6:04:5d:f4:ca:
                    cd:4b:75:78:a9:f8:a7:3b:5f:8e:de:b1:ce:38:89:
                    17:b0:b5:ef:1e:0d:7a:02:0c:58:6b:cd:00:eb:bc:
                    0c:dd:e8:e0:05:a5:4f:50:6d:60:4d:ff:f7:fe:e1:
                    5c:c4:72:f1:96:7f:7f:f0:b6:0f:68:44:55:26:77:
                    6a:9d:b9:65:17:1e:21:d9:c4:b6:2f:07:71:da:c0:
                    dd:57:24:e7:de:53:05:2c:62:fd:3f:42:89:8b:a2:
                    d0:0f
                Exponent: 65537 (0x10001)
        X509v3 extensions:
            X509v3 Subject Key Identifier: 
                8B:DE:EE:4C:7C:2D:7B:93:51:2F:FD:A0:D3:53:4C:58:CA:9D:70:C2
            X509v3 Authority Key Identifier: 
                keyid:CB:8C:C7:B6:85:79:A8:23:A6:CB:15:AB:17:50:2F:E6:65:43:5D:E8

            X509v3 Basic Constraints: 
                CA:FALSE
            X509v3 Key Usage: 
                Digital Signature, Key Encipherment
            X509v3 CRL Distribution Points: 

                Full Name:
                  URI:http://crl.aisec.fraunhofer.de/ids.crl

            Authority Information Access: 
                CA Issuers - URI:http://downloads.aisec.fraunhofer.de/rootcacert2016.cert

            X509v3 Extended Key Usage: 
                TLS Web Client Authentication, TLS Web Server Authentication
            X509v3 Subject Alternative Name: 
                DNS:localhost
    Signature Algorithm: sha256WithRSAEncryption
         3d:ef:38:f9:83:ae:2d:10:e3:ee:63:fd:ec:08:3c:a4:b8:36:
         35:d9:84:7e:f8:84:c6:67:00:bd:94:e8:6c:d8:3c:94:5f:b7:
         47:31:eb:20:68:48:66:d5:61:c5:36:c4:6e:63:38:73:c7:28:
         37:74:15:0c:f9:3d:ba:92:2d:2b:dc:cc:1c:92:ff:e9:06:cc:
         34:e5:41:0f:12:98:4f:95:36:a8:be:78:99:75:ca:d4:16:99:
         22:73:dc:9b:7e:8a:1b:b1:40:f5:30:a7:49:ab:bd:80:42:f9:
         95:1e:d5:d9:45:66:d6:8b:db:44:a4:fc:37:ad:48:5a:34:8f:
         08:b4:bf:8d:e3:42:3b:e4:d9:c3:a4:3b:68:b0:99:2b:0a:17:
         b2:b3:b8:63:76:c8:3b:42:fc:78:8e:7e:0d:d4:e6:7a:85:dd:
         b4:71:2c:26:94:b2:44:91:e4:df:27:70:c3:c9:0d:3c:e1:e9:
         cc:e3:ec:9b:31:59:54:2b:f8:db:83:d9:fc:e8:48:fb:5f:61:
         fd:84:8e:3c:96:09:84:63:b0:33:48:1e:c7:70:af:0e:78:86:
         4e:13:2a:b7:c2:c1:52:a9:bc:bb:cd:c2:9c:12:cd:9a:a3:8e:
         94:c2:69:9a:d4:d3:24:8c:b0:32:64:33:73:7e:7f:bc:54:a2:
         5d:7f:3f:bd
-----BEGIN CERTIFICATE-----
MIIEQzCCAyugAwIBAgICANAwDQYJKoZIhvcNAQELBQAwQDELMAkGA1UEBhMCREUx
EzARBgNVBAoMCkZyYXVuaG9mZXIxHDAaBgNVBAMME0lEUyBUZXN0IFN1YkNBIDIw
MTgwHhcNMjEwODE2MTE1MTU3WhcNMjMwODE2MTE1MTU3WjBlMQswCQYDVQQGEwJE
RTEMMAoGA1UECgwDSURTMRkwFwYDVQQLDBBQbHVnZmVzdCAyMDIxIDA4MS0wKwYD
VQQDDCQ2YzE3Mzc5MS1lZGE3LTQ2YTItOWJhNy1mNWI2ZWIzNmQ2YjEwggEiMA0G
CSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQDA/VoOKHcPPfkeKA0eggvWfhQqFbyk
P8vDQJ+Rto4Xm42Rn4F9Mtc+wEfqOahfe9OHTzJk7IBvPGHmFkchH3V9TMCsVb2I
saEzifx4YAOY/GeFQ+03L6mAh7BopwvoWpDpt3unQjzO3Iq4B6AsH0lL9jLFgZlx
d8rgWJ2YuYn58WSiMcjBVmuWdm6rM1fUCKQfXAUK/gFRPikDGOpLcIO2BF30ys1L
dXip+Kc7X47esc44iRewte8eDXoCDFhrzQDrvAzd6OAFpU9QbWBN//f+4VzEcvGW
f3/wtg9oRFUmd2qduWUXHiHZxLYvB3HawN1XJOfeUwUsYv0/QomLotAPAgMBAAGj
ggEgMIIBHDAdBgNVHQ4EFgQUi97uTHwte5NRL/2g01NMWMqdcMIwHwYDVR0jBBgw
FoAUy4zHtoV5qCOmyxWrF1Av5mVDXegwCQYDVR0TBAIwADALBgNVHQ8EBAMCBaAw
NwYDVR0fBDAwLjAsoCqgKIYmaHR0cDovL2NybC5haXNlYy5mcmF1bmhvZmVyLmRl
L2lkcy5jcmwwVAYIKwYBBQUHAQEESDBGMEQGCCsGAQUFBzAChjhodHRwOi8vZG93
bmxvYWRzLmFpc2VjLmZyYXVuaG9mZXIuZGUvcm9vdGNhY2VydDIwMTYuY2VydDAd
BgNVHSUEFjAUBggrBgEFBQcDAgYIKwYBBQUHAwEwFAYDVR0RBA0wC4IJbG9jYWxo
b3N0MA0GCSqGSIb3DQEBCwUAA4IBAQA97zj5g64tEOPuY/3sCDykuDY12YR++ITG
ZwC9lOhs2DyUX7dHMesgaEhm1WHFNsRuYzhzxyg3dBUM+T26ki0r3Mwckv/pBsw0
5UEPEphPlTaovniZdcrUFpkic9ybfoobsUD1MKdJq72AQvmVHtXZRWbWi9tEpPw3
rUhaNI8ItL+N40I75NnDpDtosJkrCheys7hjdsg7Qvx4jn4N1OZ6hd20cSwmlLJE
keTfJ3DDyQ084enM4+ybMVlUK/jbg9n86Ej7X2H9hI48lgmEY7AzSB7HcK8OeIZO
Eyq3wsFSqby7zcKcEs2ao46Uwmma1NMkjLAyZDNzfn+8VKJdfz+9
-----END CERTIFICATE-----
```

**Review by denisneuling — APPROVED** [review]

File: extensions/iam/daps/src/test/resources/keys/OEI6REU6RUU6NEM6N0M6MkQ6N0I6OTM6NTE6MkY6RkQ6QTA6RDM6NTM6NEM6NTg6Q0E6OUQ6NzA6QzI6a2V5aWQ6Q0I6OEM6Qzc6QjY6ODU6Nzk6QTg6MjM6QTY6Q0I6MTU6QUI6MTc6NTA6MkY6RTY6NjU6NDM6NUQ6RTg=.cert (hunk 1 above)
Comment (ndr-brt) [reply extensions/iam/daps/src/test/resources/keys/OEI6REU6RUU6NEM6N0M6MkQ6N0I6OTM6NTE6MkY6RkQ6QTA6RDM6NTM6NEM6NTg6Q0E6OUQ6NzA6QzI6a2V5aWQ6Q0I6OEM6Qzc6QjY6ODU6Nzk6QTg6MjM6QTY6Q0I6MTU6QUI6MTc6NTA6MkY6RTY6NjU6NDM6NUQ6RTg=.cert:25]: I extracted it from the `plugfest-keystore.p12` that's contained in the test resources but yes, maybe would be better to create one with a longer validity.

Comment (ndr-brt) [reply extensions/iam/daps/src/test/resources/keys/OEI6REU6RUU6NEM6N0M6MkQ6N0I6OTM6NTE6MkY6RkQ6QTA6RDM6NTM6NEM6NTg6Q0E6OUQ6NzA6QzI6a2V5aWQ6Q0I6OEM6Qzc6QjY6ODU6Nzk6QTg6MjM6QTY6Q0I6MTU6QUI6MTc6NTA6MkY6RTY6NjU6NDM6NUQ6RTg=.cert]: Ok, I created and introduced another certificate that expires in 2031

**Review by paullatzelsperger — APPROVED** [review]
It works locally, so I think we can merge it. Should it not work in CI, we can always open a bug issue. Generally I think we should improve the test coverage in a subsequent issue - there is only 1 test method now.

Also, I think putting certificates (or rather: their contents) into a repository secret seems cleaner, but it's not a showstopper either.

Just one last thing to note - I ran into a file system permission issue in my machine. This will not necessarily affect the Github runner, just something to keep in mind.

**Comment by ndr-brt** [comment]
ready for merge

**Pull request merged**